README.md

```diff
@@ -1132,6 +1132,40 @@ If we used `@append_spaced_softline` rather than
 `@append_spaced_scoped_softline`, the `1,` would be followed by a space rather
 than a newline, because it's inside a single-line `product_expression`.
 
+#### `@prepend_begin_measuring_scope` / `@append_begin_measuring_scope` / `@prepend_end_measuring_scope` / `@append_end_measuring_scope`
+
+Sometimes, custom scopes are not enough: we may want format a node depending on the multi-line-ness of a piece of code that does not include the node in question. For instance, consider this function application in OCaml:
+```ocaml
+foo bar (fun x -> qux)
+```
+We may also want to format it as any of the following two, depending on the actual length of `foo`, `bar`, and `qux`:
+```ocaml
+foo bar (fun x ->
+  qux
+)
+```
+```ocaml
+foo
+  bar
+  (fun x ->
+    qux
+  )
+```
+Consider the indentation of `(fun x -> qux)`: if `foo bar` is single-line, we don't want to indent it. But if `foo bar` is multi-line, we do want to indent it.
+
+Because custom scopes can only impact the behaviour of nodes inside the scope, we can't use them to solve this issue. This is why we need `measuring_scope`.
+
+Measuring scopes are opened/closed with a similar syntax as "regular" custom scopes, with any of the following tags, in conjunction with the `#scope_id!` predicate:
+
+* `@prepend_begin_measuring_scope`
+* `@append_begin_measuring_scope`
+* `@prepend_end_measuring_scope`
+* `@prepend_begin_measuring_scope`
+
+The measuring scopes behave as follows:
+* A measuring scope must always be contained in a regular custom scope with the same `#scope_id!`. There can't be two measuring scopes with the same `#scope_id!` inside the same regular custom scope.
+* If a regular custom scope contains a measuring scope, then all tags contained in the regular scope that depend on its multi-line-ness will instead depend on the multi-line-ness of the measuring scope (hence its name: the inner, measuring scope measures the multi-line-ness of the outer, regular scope).
```

Comment:
With this description, I don't understand why

```ocaml
foo bar (fun x ->
  qux)
```

isn't formatted as

```ocaml
foo
  bar
  (fun x ->
    qux
  )
```

I think I'm missing the point.

Reply:
I believe the measuring scope is in this case `foo bar (` (or maybe just `bar (`, but the point is that it stops at `(`). Which means that the whole scope, although it spans multiple lines, is considered to be single line, and the application isn't laid out on multiple lines.